hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContract.php

```diff
@@ -193,21 +194,26 @@ protected static function getContractDates($contracts) {
     // If there are two (or more) Contracts starting on the same day
     // then we pick only the one with latest End date.
     foreach ($contracts['values'] as $contract) {
-      $details = civicrm_api3('HRJobDetails', 'getsingle', array(
+      $details = civicrm_api3('HRJobDetails', 'get', array(
         'sequential' => 1,
```

Comment:
also I was thinking maybe you can remove this API call from **calculateLengthOfService()** :

```php
    // Getting all Job Contracts for given Contact ID.
    $contracts = civicrm_api3('HRJobContract', 'get', array(
      'sequential' => 1,
      'contact_id' => (int)$contactId,
      'deleted' => 0,
      'options' => array('limit' => 0),
    ));
```

and here inside **getContractDates()** you can call **getContactContracts()** and then loop through the results here  ... it may be also a good chance to improve **getContactContracts()** and make it depend on BAO methods instead of API since.